### PR TITLE
Reconfigure the `chromosomal-instability` analysis code

### PR DESCRIPTION
**Pull request by cansavvy**
#### What scientific question is your analysis addressing?
This was not a new question but code reconfiguration associated with the [issues discussed in this comment](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/394#issuecomment-576725226) on #394. 

#### What was your approach?
Here are the changes made in this PR: 

1) The BED files have been wrangled for calculating `break_density`. You will note there's one spot we need to switch out the BED file for when we have the final list of unsurveyed genome regions.  

2) What was a long notebook has now been reconfigured to a script that takes options and now the notebook only does the plots. This was done so we can easily switch in and out files as well as run this data for its output which will be used in places other than the notebook of this analysis. 

3) The output of calculations is now saved to a set of [TSV files so that one of the subtyping tickets can use this information](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/245#issuecomment-576710898). 

| File | Description | Relevant origin files |
|-----|-------------|---------------------|
| `analyses/chromosomal-instability/run_breakpoint_analysis.sh` | Shell script for running analysis module |  Calls the 00-script and notebook. [Has borrowed the BEDtools element from here.](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/fa214291713575be7fd20c92374b268870f4173f/analyses/snv-callers/run_caller_consensus_analysis.sh#L66) |
| `analyses/chromosomal-instability/plot-chromosomal-instability.Rmd` | Has the plotted pieces of the original notebook. | [The second half of the chromosomal-plots notebook](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/fa214291713575be7fd20c92374b268870f4173f/analyses/chromosomal-instability/chromosomal-instability.Rmd#L210) |
| `analyses/chromosomal-instability/00-setup-breakpoint-data.R` | Sets up the data into "breakpoint" format and saves data files that are consumed by the notebook. | [Is from the first half of the original notebook](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/fa214291713575be7fd20c92374b268870f4173f/analyses/chromosomal-instability/chromosomal-instability.Rmd#L67) the last part of the script are [elements are borrowed from  tmb-calculations](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/fa214291713575be7fd20c92374b268870f4173f/analyses/snv-callers/util/tmb_functions.R#L117).  | |

#### What GitHub issue does your pull request address?

https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/394#issuecomment-576725226

#### Which areas should receive a particularly close look?
- Does this help set us up for addressing the [goals laid out in this comment](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/394#issuecomment-576725226)?
- Do the options for the script run as expected? Are there other options you would expect to see? \

**NOTE** GitHub isn't registering that most of the code in the 00 script is the moved from the notebook. You shouldn't have to do an in-depth review of that whole script because a lot of it is the same as before, just in a different location. 

#### Is there anything that you want to discuss further?
Regrettably, there are many things hardcoded into the CNV and SV file wrangling. So the `cnv_seg` and `sv` options are not as flexible as I would like. Do you have recommendations for this that aren't too into the weeds but would have a good payoff for making this more generalizable?

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
There are output files that can be looked at in the `breakpoint-data` file. e.g. `cnv_breaks_densities.tsv`. 
Do these files have all the information you would need and expect? 

The notebook can be seen here, but it is essentially the same output: https://cansavvy.github.io/openpbta-notebook-concept/chromosomal-instability/plot-chromosomal-instability.nb.html

#### Reproducibility Checklist

These items were addressed in a previous PR. 
- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/chromosomal-instability/00-setup-breakpoint-data.R [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+  dplyr::filter(
+    !(chrom %in% c("X", "Y", "M", NA)),
+    # Only keep samples that have both CNV and SV data
+    samples %in% common_samples
+  )
+
+############################## Find common breaks ##############################
+# Make an common breaks data.frame.
+common_breaks <- dplyr::bind_rows(sv_breaks, cnv_breaks) %>%
+  dplyr::distinct(samples, chrom, coord, .keep_all = TRUE)
```
[the remark this answers is not in the document]
Comment (cansavvy) [reply analyses/chromosomal-instability/00-setup-breakpoint-data.R]: Ah. Good point. I can change this to be something more specific. `both_breaks`? Also side note, I tried intersection and only about 36 breaks out of all samples' data are the same exact call between SV and CNV (this is using exact coordinates). I talked to @jashapiro about this and he said this wasn't surprising and that perhaps in a future PR we could play around with being more flexible with the calls and whether we allow flexibiltty in the coordinates (ie. +_ how many bp away do we consider it the same).

**Comment by cansavvy** [comment]
Side note on the numbering, why I didn't end up calling this `01-` automatically was because when I was originally planning this out, I was planning on the recurrent genes' notebooks thing being their own notebooks which would be in a parallel order to this. aka, either could be run after the `00` notebook. But that being said, I'll add the `01` numbering for now, and then when I add the later notebooks, maybe make them `01a` and `01b` or something.

File: analyses/chromosomal-instability/00-setup-breakpoint-data.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply analyses/chromosomal-instability/00-setup-breakpoint-data.R]: `union_of_breaks` would be pretty clear?

[the remark this answers is not in the document]
Comment (cansavvy) [reply analyses/chromosomal-instability/00-setup-breakpoint-data.R]: ^I'll convert it to that.

**Comment by cansavvy** [comment]
I didn't find anything that would give me clues as to CBS algorithm derived data having the same. I think I'll just leave this as is and if we end up using it elsewhere and find ways to make it more generalizable then we can do it then.

**Comment by cansavvy** [comment]
Alrighty, @jaclyn-taroni. I addressed your comments. This is ready for another gander.

**Pull request merged**